tools/ci/docker/linux/Dockerfile

```diff
@@ -381,6 +381,10 @@ ENV PATH="/tools/gcc-arm-none-eabi/bin:$PATH"
 COPY --from=nuttx-toolchain-arm64 /tools/gcc-aarch64-none-elf/ gcc-aarch64-none-elf/
 ENV PATH="/tools/gcc-aarch64-none-elf/bin:$PATH"
 
+# AVR32 toolchain
+COPY --from=nuttx-toolchain-avr32 /tools/avr32-gnu-toolchain-linux_x86/ avr32-gnu-toolchain-linux_x86/
```

Comment:
where you download the toolchain?

Reply:
Dear @xiaoxiang781216,

> where you download the toolchain?

I intend to download the toolchain from [1]. In the code, where must I
add it so CI can see it? I have added some code in [2] which clones
the toolchain. Am I missing something? Here is a snip from [2]:

```
avr32_gcc_toolchain() {
  add_path "${NUTTXTOOLS}"/avr32-gnu-toolchain-linux_x86/bin

  if [ ! -f "${NUTTXTOOLS}/avr32-gnu-toolchain-linux_x86/bin/avr32-gcc" ]; then
    local basefile
    basefile=avr32-gnu-toolchain-linux_x86
    cd "${NUTTXTOOLS}"
    # Download the AVR32 GCC toolchain
    git clone https://github.com/ramangopalan/${basefile}
  fi

  command avr32-gcc --version
}
```

References:
[1]: https://github.com/ramangopalan/avr32-gnu-toolchain-linux_x86
[2]: [tools/ci/platforms/ubuntu.sh](https://github.com/apache/nuttx/pull/11709/commits/044080c6e3e692ad78f5595ab7c5cbc5fcab5d82)

Reply:
@xiaoxiang781216 could you please help here?

Reply:
@ramangopalan need to add in the Dockerfile:

 https://github.com/apache/nuttx/blob/4178f3ede4b0be561ec7e34b049b48d69707ff65/tools/ci/docker/linux/Dockerfile#L120
```` 
###############################################################################
# Build image for tool required by AVR32 builds
###############################################################################
FROM nuttx-toolchain-base AS nuttx-toolchain-avr32
# Download the latest AVR32 GCC toolchain prebuilt by Raman
RUN mkdir -p gcc-avr32-gnu && \
  git clone https://github.com/ramangopalan/avr32-gnu-toolchain-linux_x86 gcc-avr32-gnu
````
https://github.com/apache/nuttx/blob/4178f3ede4b0be561ec7e34b049b48d69707ff65/tools/ci/docker/linux/Dockerfile#L385
````
# AVR32 toolchain
COPY --from=nuttx-toolchain-avr32 /tools/gcc-avr32-gnu/ gcc-avr32-gnu/
ENV PATH="/tools/gcc-avr32-gnu/bin:$PATH"
````
https://github.com/apache/nuttx/blob/4178f3ede4b0be561ec7e34b049b48d69707ff65/tools/ci/docker/linux/Dockerfile#L440
````
  ln -sf `which ccache` /tools/ccache/bin/avr32-gcc && \
  ln -sf `which ccache` /tools/ccache/bin/avr32-g++ && \
````

Reply:
Thank you very much @simbit18 !!!

@ramangopalan could you please include these suggestion? If it fixes the issue we could integrate this PR and get AVR32 tested in our CI

Reply:
Dear @simbit18, Dear @acassis,

Many thanks for your suggestion. I will give these a try in just a
jiffy and write back.

Reply:
Not sure why clone didn't go though. Did I miss something?
Could it be about a setting on Github?

```
--------------------
 125 |     # Download the prebuilt AVR32 GCC toolchain
 126 | >>> RUN mkdir -p gcc-avr32-gnu && \
 127 | >>>   git clone https://github.com/ramangopalan/avr32-gnu-toolchain-linux_x86 gcc-avr32-gnu
 128 |     
--------------------
ERROR: failed to solve: process "/bin/sh -c mkdir -p gcc-avr32-gnu &&   git clone https://github.com/ramangopalan/avr32-gnu-toolchain-linux_x86 gcc-avr32-gnu" did not complete successfully: exit code: 127
```

Reply:
@ramangopalan in the stage is missing git try this:
````
###############################################################################
# Build image for tool required by AVR32 builds
###############################################################################
FROM nuttx-toolchain-base AS nuttx-toolchain-avr32
# Download the prebuilt AVR32 GCC toolchain
RUN apt-get update -qq && DEBIAN_FRONTEND="noninteractive" apt-get install -y -qq \
  git
# Clone Main Repository 
RUN mkdir -p gcc-avr32-gnu && \
  git clone --depth 1 https://github.com/ramangopalan/avr32-gnu-toolchain-linux_x86 gcc-avr32-gnu
````